tests/python/pants_test/backend/jvm/tasks/jvm_compile/zinc/zinc_compile_integration_base.py

```diff
@@ -343,6 +343,51 @@ def test_zinc_logs_warnings_properly(self):
         for expected in expected_strings:
             self.assertIn(expected, pants_run.stdout_data)
 
+    def _test_zinc_reports_diagnostic_counts(self, reporting):
+        with self.temporary_workdir() as workdir:
+            target = "testprojects/src/scala/org/pantsbuild/testproject/compilation_warnings/unused_import_warning:unused_import"
+            with self.temporary_cachedir() as cachedir:
+                args = ['--compile-rsc-args=+["-S-Ywarn-unused:_"]'] + (
+                    ["--compile-rsc-report-diagnostic-counts"] if reporting else []
+                )
+                pants_run = self.run_test_compile(workdir, cachedir, target, extra_args=args,)
+                self.assert_success(pants_run)
+
+            expected_strings = [
+                "Reporting number of diagnostics for: ScalaLibrary(testprojects/src/scala/org/pantsbuild/testproject/compilation_warnings/unused_import_warning:unused_import)",
+                "Error: 0",
+                "Warning: 1",
+                "Information: 0",
+                "Hint: 0",
+            ]
+
+            for expected in expected_strings:
+                if reporting:
+                    self.assertIn(expected, pants_run.stdout_data)
```

Comment:
nit: You could use `dedent("\n".join(expected_strings))` to find the entire text, instead of line by line. I don't think it matters in this case, but if two targets have diagnostics, this series of asserts won't check order, giving false results.

Reply:
I don't think it matters in this case :)

Reply:
That's fair enough, but since I'm only testing with one target, I'd rather keep my precious green CI than edit this in this case (except if some other review comments forces me to edit the PR once more anyway)

Reply:
Uhm. I played with this and it's not that straightforward because using dedent still is sensitive to the indentation. I had rather not couple this test to the indentation of the output, so I'll just keep it as is.